File: src/backend/studio-reservation/src/main/java/org/taonaw/studio_reservation/domain/model/staffAccount/StaffAccountRepository.java
package org.taonaw.studio_reservation.domain.model.staffAccount;

import java.util.Optional;

public interface StaffAccountRepository {
    Optional<StaffAccount> findBy(LoginId loginId);
    AddResults add(StaffAccount staffAccount);

    enum AddResults {
        SUCCEEDED,
        DUPLICATED;

        public boolean isSucceeded() {
            return this == SUCCEEDED;
        }
    }
}

File: src/backend/identityaccess/src/main/kotlin/org/taonaw/identityaccess/domain/model/staffAccount/StaffAccountRepository.kt
package org.taonaw.identityaccess.domain.model.staffAccount

interface StaffAccountRepository {
    fun findBy(staffAccountId: StaffAccountId): StaffAccount?
    fun findBy(userId: UserId): StaffAccount?
    fun save(staffAccount: StaffAccount): SaveResult

    enum class SaveResult {
        SUCCEEDED,
        USER_ID_ALREADY_REGISTERED
    }
}